src/OpenTelemetry/Metrics/MeterProviderBuilderBase.cs

```diff
@@ -88,19 +89,50 @@ internal MeterProviderBuilder AddReader(MetricReader reader)
 
         internal MeterProviderBuilder AddView(string instrumentName, string name)
         {
-            // TODO: Actually implement view.
-            return this;
+            var metricConfig = new MetricStreamConfiguration() { Name = name };
+            return this.AddView(instrumentName, metricConfig);
         }
 
-        internal MeterProviderBuilder AddView(string instrumentName, MetricStreamConfiguration aggregationConfig)
+        internal MeterProviderBuilder AddView(string instrumentName, MetricStreamConfiguration metricStreamConfiguration)
         {
-            // TODO: Actually implement view.
-            return this;
+            Func<Instrument, MetricStreamConfiguration> viewConfig;
+            if (instrumentName.IndexOf('*') != -1)
+            {
+                var pattern = '^' + Regex.Escape(instrumentName).Replace("\\*", ".*");
+                var regex = new Regex(pattern, RegexOptions.Compiled | RegexOptions.IgnoreCase);
+                viewConfig = (instrument) =>
+                {
+                    if (regex.IsMatch(instrument.Name))
+                    {
+                        return metricStreamConfiguration;
+                    }
+                    else
+                    {
+                        return null;
+                    }
+                };
+            }
+            else
+            {
+                viewConfig = (instrument) =>
+                {
+                    if (instrument.Name.Equals(instrumentName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return metricStreamConfiguration;
+                    }
+                    else
+                    {
+                        return null;
+                    }
+                };
+            }
+
+            return this.AddView(viewConfig);
         }
 
         internal MeterProviderBuilder AddView(Func<Instrument, MetricStreamConfiguration> viewConfig)
```

Comment:
AggressivelyInline

Reply:
not in the perf-critical instrument creation path. so won't be needed.